cpu/nrf51/periph/spi.c

```diff
@@ -21,216 +21,113 @@
-            break;
-        case SPI_CONF_SECOND_RISING:
-            spi[dev]->CONFIG = (SPI_CONFIG_CPOL_ActiveHigh << 2) | (SPI_CONFIG_CPHA_Trailing << 1);
-            break;
-        case SPI_CONF_FIRST_FALLING:
-            spi[dev]->CONFIG = (SPI_CONFIG_CPOL_ActiveLow << 2) | (SPI_CONFIG_CPHA_Leading << 1);
-            break;
-        case SPI_CONF_SECOND_FALLING:
-            spi[dev]->CONFIG = (SPI_CONFIG_CPOL_ActiveLow << 2) | (SPI_CONFIG_CPHA_Trailing << 1);
-            break;
-    }
-
-    /* select bus speed */
-    switch (speed) {
-        case SPI_SPEED_100KHZ:          /* 125 KHz for this device */
-            spi[dev]->FREQUENCY = SPI_FREQUENCY_FREQUENCY_K125;
-            break;
-        case SPI_SPEED_400KHZ:          /* 500 KHz for this device */
-            spi[dev]->FREQUENCY = SPI_FREQUENCY_FREQUENCY_K500;
-            break;
-        case SPI_SPEED_1MHZ:            /* 1 MHz for this device */
-            spi[dev]->FREQUENCY = SPI_FREQUENCY_FREQUENCY_M1;
-            break;
-        case SPI_SPEED_5MHZ:            /* 4 MHz for this device */
-            spi[dev]->FREQUENCY = SPI_FREQUENCY_FREQUENCY_M4;
-            break;
-        case SPI_SPEED_10MHZ:           /* 8 MHz for this device */
-            spi[dev]->FREQUENCY = SPI_FREQUENCY_FREQUENCY_M8;
-            break;
-    }
+static mutex_t locks[SPI_NUMOF];
 
-    /* finally enable the device */
-    spi[dev]->ENABLE = 1;
-    return 0;
-}
-
-int spi_init_slave(spi_t dev, spi_conf_t conf, char (*cb)(char data))
+static inline NRF_SPI_Type *dev(spi_t bus)
 {
-    (void) dev;
-    (void) conf;
-    (void) cb;
-    /* This API is incompatible with nRF51 SPIS */
-    return -1;
+    return spi_config[bus].dev;
 }
 
-int spi_conf_pins(spi_t dev)
+void spi_init(spi_t bus)
 {
-    switch (dev) {
-#if SPI_0_EN
-        case SPI_0:
-            /* set pin direction */
-            NRF_GPIO->DIRSET = (1 << SPI_0_PIN_MOSI) | (1 << SPI_0_PIN_SCK);
-            NRF_GPIO->DIRCLR = (1 << SPI_0_PIN_MISO);
-            /* select pins to be used by SPI */
-            spi[dev]->PSELMOSI = SPI_0_PIN_MOSI;
-            spi[dev]->PSELMISO = SPI_0_PIN_MISO;
-            spi[dev]->PSELSCK = SPI_0_PIN_SCK;
-            break;
-#endif
-#if SPI_1_EN
-        case SPI_1:
-            /* set pin direction */
-            NRF_GPIO->DIRSET = (1 << SPI_1_PIN_MOSI) | (1 << SPI_1_PIN_SCK);
-            NRF_GPIO->DIRCLR = (1 << SPI_1_PIN_MISO);
-            /* select pins to be used by SPI */
-            spi[dev]->PSELMOSI = SPI_1_PIN_MOSI;
-            spi[dev]->PSELMISO = SPI_1_PIN_MISO;
-            spi[dev]->PSELSCK = SPI_1_PIN_SCK;
-            break;
-#endif
-        default:
-            return -1;
-    }
-    return 0;
-}
+    assert(bus < SPI_NUMOF);
 
-int spi_acquire(spi_t dev)
-{
-    if ((unsigned int)dev >= SPI_NUMOF) {
-        return -1;
-    }
-    mutex_lock(&locks[dev]);
-    return 0;
+    /* initialize mutex */
+    mutex_init(&locks[bus]);
```

Comment:
IMHO no need. The `spi_init()` funciton is supposed to be called once during system startup. At that point, there is no one alive (e.g. not threads started yet) who could use the device, so hence need for locking in the init function.